test/mocha/11-keystore-kms-client.js

```diff
@@ -56,10 +56,10 @@ describe('keystore API interactions using webkms-client', () => {
     should.not.exist(result);
     should.exist(err);
     err.status.should.equal(400);
-    err.data.should.have.keys('message', 'type', 'details', 'cause');
+    err.data.should.include.keys('message', 'type', 'details', 'cause');
     err.data.type.should.equal('URLMismatchError');
-    err.data.details.should.have.keys(
-      ['configId', 'expected', 'httpStatusCode', 'actual', 'requestUrl']);
+    err.data.details.should.include.keys(
```

Comment:
-1 to this change.  Typically, these assertions are designed to ensure that there is no extraneous data in values.  Any changes to the error value *should* require a conscious intentional change to the test.

Reply:
@mattcollier, the `have` approach means our test suites break when we release minor releases. Is that what we want?

Reply:
Or maybe there's a middle ground -- where `have` should be used for in-module generated errors and `include` on errors generated from a dependency?

Reply:
Historically, we have not been careful in our treatment of error data models/messages as breaking changes.  I think there a good case to be made that if there is a change (local or subdependency) that changes the public errors surface here, that should be treated as a breaking change.

The better our test coverage, especially around errors downstream, the more it will become evident that we are/aren't doing this right.

Reply:
So, I'm +1 for doing `have` in this particular case because this module generates the error. That gets this PR moving. I will push for `include` to be used in the other cases I mentioned because I think that makes more sense for extensibility and allowing minor releases.

Reply:
ack, we can discuss in another venue.